united-fare-finder.py

```diff
@@ -48,54 +48,75 @@
 
     elem = driver.find_element(
       By.ID, "bookFlightOriginInput")
+    # on macbook, ctrl+a doesn't work, so also adding clear()
+    elem.clear()
     elem.send_keys(Keys.CONTROL + "a")
+    elem.send_keys(Keys.COMMAND + "a")
     elem.send_keys(ORIGIN)
     elem.send_keys(Keys.TAB)
 
     elem = driver.find_element(By.ID, "bookFlightDestinationInput")
+    elem.clear()
     elem.send_keys(Keys.CONTROL + "a")
+    elem.send_keys(Keys.COMMAND + "a")
     elem.send_keys(DEST)
     elem.send_keys(Keys.TAB)
 
     elemList = driver.find_elements(
       By.XPATH, '//div[@class="DateInput DateInput_1"]/input')
 
     elem = elemList[0]
+    elem.clear()
     elem.send_keys(Keys.CONTROL + "a")
+    elem.send_keys(Keys.COMMAND + "a")
     elem.send_keys(DEP_DATE)
 
-    elem = elemList[1]
-    elem.send_keys(Keys.CONTROL + "a")
-    elem.send_keys(RET_DATE)
-
+    # 'Book with miles' checkbox, can't click checkbox directly
     elem = driver.find_element(
       By.XPATH, '//*[@id="bookFlightForm"]/div[1]/div[1]/label')
+    checkbox = driver.find_element(
+      By.ID, 'award')
+    # Dont click if checkbox is already checked
+    if checkbox.is_selected() != True:
+      elem.click()
+
+    # 'One-way radio button'
+    elem = driver.find_element(By.ID, 'oneway')
     elem.click()
 
+    # 'Find flights' button
     elem = driver.find_element(
       By.CSS_SELECTOR,
       "button."
       "app-components-BookFlightForm-bookFlightForm__findFlightBtn--1lbFe")
     elem.submit()
+    
+    elem = driver.find_element(
+      By.CSS_SELECTOR,
+      "button."
+      "app-components-Shopping-Calendar-styles__calendarIcon--2nO38")
+    elem.click()
 
-    time.sleep(15)
+    # Waits for calendar to load on page
+    elem = driver.find_element(
+      By.CSS_SELECTOR,
+      "div."
+      "app-components-Shopping-FlexibleCalendar-styles__content--11v1q")
 
     # scrape fare data
     page_source = driver.page_source
     soup = BeautifulSoup(page_source, "html.parser")
-    flight_elements = parsePageSource(soup)
+    price_elements = soup.find_all(
+        "div",
+        class_="app-components-Shopping-FlexibleCalendar-styles__content--11v1q")
+   
 
     # write to file
-    print(DEST + " ********************************************************\n")
-    for i in range(len(flight_elements)):
-        for j in range(len(flight_elements[i])):
-            print(*flight_elements[i][j])
-        print()
 
-    # return to home page
-    elem = driver.find_element(By.ID, "unitedLogo")
-    elem.click()
+    for element in price_elements:
+      print(ORIGIN+", ", DEST+", ", element.text)
```

Comment:
element.text just strips out all of the text within the div.  I think it will be easier to parse the resulting string into CSV rather than dealing with the html. But I could be convinced otherwise. 

The string output here is pretty close to CSV, I think with some regex we could get it there pretty easily